### ISSUE TITLE
Allow for fixup! commits

### ISSUE THREAD

**Issue by blaggacao** [siderolabs/conform#225]
When using conventional commits, allow the use of `fixup! <sha>` or `fixup! <header>` syntax, which are both recognized by `git rebase --autosquash` later on.

**Comment by sdurrheimer** [comment]
Agreed, `fixup!` and `squash!` prefixes should be ignored as those mechanisms are part of the base features Git offers.

**Comment by janderssonse** [comment]
@blaggacao I did'n try this, but I think this could be solved by a setting in the .conform.yml, just adding allowed types:

`
    conventional:
      types: ["fixup!,"wip", "chore","docs","perf","refactor","style","test","release"]
      scopes: [".*"]
`

**Comment by blaggacao** [comment]
That's a good idea, however we eneded up wrapping, with regex condition.

That hives more flexibility to the actual header and thereby less restriction to the end use: https://github.com/divnix/std/commit/2f047e0707ef15e89582d51e46dec796b582a520

**Comment by janderssonse** [comment]
Any way that works is the right way! :) Sidenote, I really like this tool, but would love to see more documentation, and more responses to the issues from the maintainer.

**Comment by terlar** [comment]
The type `fixup!` and `squash!` won't work since it is in the format of `fixup! previous commit message` so it doesn't follow the format, it would have to be `fixup!:`. It would be nice with support for this without having to wrap conform in a shell script.

**Comment by qqii** [comment]
Any chance for this to be reopened? I'd love to see the following supported:

```
fixup! <commit_ish>
squash! <commit_ish>
amend! <commit_ish>
```

Adding them to the `types` is not enough as [the regex](https://github.com/siderolabs/conform/blob/53b06192ca4b9bbeac3f89299a2a2eb792a63ce3/internal/policy/commit/check_conventional_commit.go#L25) enforces a colon:

```go
var HeaderRegex = regexp.MustCompile(`^(\w*)(\(([^)]+)\))?(!)?:\s{1}(.*)($|\n{2})`)
```

Edit: I've made an attempt at implementing this in #282.
